src/integrations/prefect-dbt/prefect_dbt/cli/credentials.py

```diff
@@ -152,3 +167,16 @@ def get_profile(self) -> Dict[str, Any]:
             },
         }
         return profile
+
+    @field_validator("target_configs", mode="before")
+    @classmethod
+    def handle_target_configs(cls, v: Any) -> Any:
+        """Handle target configs field aliasing during validation"""
+        if isinstance(v, dict):
+            if "schema_" in v:
+                v["schema"] = v.pop("schema_")
+            # Handle nested blocks
+            for value in v.values():
+                if isinstance(value, dict) and "schema_" in value:
+                    value["schema"] = value.pop("schema_")
+        return v
```

Comment:
This validator feels like it should be on `BaseTargetConfigs` instead. I think you wouldn't have to worry about nested blocks in that case.

Reply:
@desertaxle i mentioned this [here](https://github.com/PrefectHQ/prefect/pull/16846#discussion_r1928947875), maybe it should be a before `model_validator` instead of `field_validator`? bc the alias `schema_` presents an issue with the `field_validator` I think

Reply:
For posterity, @zzstoatzz and I looked at this quite thoroughly, and it seems like we are inconsistent in our use of aliases when dumping Blocks. We agreed that the current validator approach is a good short term option and we'll work on adding additional validation aliases to `schema_` fields to ensure we handle the inconsistency in Block dumping.